pkg/kapp/resources/resource_types.go

```diff
@@ -269,3 +269,16 @@ func NonMatching(in []ResourceType, ref ResourceRef) []ResourceType {
 	}
 	return out
 }
+
+// TODO: Extend ResourceRef and PartialResourceRefd to allow GVK matching
+func MatchingAnyGK(in []ResourceType, gks []schema.GroupKind) []ResourceType {
+	var out []ResourceType
+	for _, item := range in {
+		for _, gk := range gks {
+			if (GKResourceRef{gk}).Matches(item) {
```

Comment:
That was my original intention, but `ResourceType` does not seem to expose that function.
Would be prefer constructing a `GroupKind` struct here and passing it over passing a `ResourceType` struct?
(I could implement the function but chose to not do that as the other functions surfacing group, kind, etc are due to the structs embedded in `ResourceType`)

Reply:
> GroupKind struct here and passing it over passing a ResourceType struct?

We can do that. I don't have any strong reason apart from the fact that the other information is unnecessary/noisy in `Matches()`, so I will it up to you.

Reply:
That is reasonable. I however did that in favour of keeping the function `MatchinAnyGK` concise.
I will leave this thread unresolved in case anyone wants to add any thoughts 🤔